components/collapsible-panel/collapsible-panel.js

```diff
@@ -0,0 +1,405 @@
+				@focus="${this._onFocus}"
+				@blur="${this._onBlur}"
+			>${this.expandCollapseLabel}</button>
+			<div class="d2l-collapsible-panel" @click="${this._handlePanelClick}">
+				<div class="d2l-collapsible-panel-top-sentinel"></div>
+				${this._renderHeader()}
+				<d2l-expand-collapse-content ?expanded="${this.expanded}">
+					<div class="d2l-collapsible-panel-content">
+						<slot name="content"></slot>
+					</div>
+				</d2l-expand-collapse-content>
+				<d2l-expand-collapse-content ?expanded="${!this.expanded}">
+					<div class="d2l-collapsible-panel-summary">
+						<slot name="summary" @slotchange="${this._handleSummarySlotChange}"></slot>
+					</div>
+				</d2l-expand-collapse-content>
+			</div>
+		`;
+	}
+
+	_handleActionsClick(e) {
+		const actions = this.shadowRoot.querySelector('.d2l-collapsible-panel-header-actions');
+		if (e.target !== actions) {
+			e.stopPropagation();
+		}
+	}
+
+	_handleHeaderClick(e) {
+		if (this.expanded) {
+			this._toggleExpand();
+			e.stopPropagation();
+		}
+	}
+
+	_handleHeaderSecondaryClick(e) {
+		const header = this.shadowRoot.querySelector('.d2l-collapsible-panel-header-secondary');
+		if (e.target !== header) {
+			e.stopPropagation();
+		}
+	}
+
+	_handlePanelClick(e) {
+		if (!this.expanded) {
+			this._toggleExpand();
+			e.stopPropagation();
+		}
+	}
+
+	_handleSummarySlotChange() {
+		const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+		element.classList.add('has-summary');
+	}
+
+	_onBlur() {
+		const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+		element.classList.remove('focused');
+	}
+
+	_onFocus() {
+		const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+		element.classList.add('focused');
+	}
+
+	_renderHeader() {
+		return html`
+			<div class="d2l-collapsible-panel-header" @click="${this._handleHeaderClick}">
+				<div class="d2l-collapsible-panel-header-primary">
+					${this._renderHeading()}
+					<div class="d2l-collapsible-panel-header-actions" @click="${this._handleActionsClick}">
+						<slot name="actions" @slotchange="${this._handleActionsSlotChange}"></slot>
+					</div>
+					<div class="d2l-collapsible-panel-opener">
+						<d2l-icon-custom size="tier1">
+							<svg xmlns="http://www.w3.org/2000/svg" width="10" height="18" fill="none" viewBox="0 0 10 18">
+								<path stroke="#494c4e" stroke-linejoin="round" stroke-width="2" d="m9 9-8 8V1l8 8Z"/>
+							</svg>
+						</d2l-icon-custom>
+					</div>
+				</div>
+				<div class="d2l-collapsible-panel-header-secondary" @click="${this._handleHeaderSecondaryClick}">
+					<slot name="header" @slotchange="${this._handleHeaderSlotChange}"></slot>
+				</div>
+			</div>
+			<div class="d2l-collapsible-panel-divider"></div>
+		`;
+	}
+
+	_renderHeading() {
+		const titleClasses = {
+			'd2l-collapsible-panel-title': true,
+			[`d2l-heading-${this.headingStyle}`]: true,
+		};
+
+		const headingLevel = parseInt(this.headingLevel);
```

Comment:
Do you want to normalize it in case they set it to something `< 1` or `> 4`? Probably similar for `headingStyle`.